src/layout/header/index.js

```diff
@@ -1,17 +1,65 @@
-import React from "react"
-// import PropTypes from "prop-types"
-import Link from "../../components/link"
+import React, { useState } from "react"
+import PropTypes from "prop-types"
+import classNames from "classnames"
+
+import Icon from "./../../components/icon"
+import Link from "./../../components/link"
+import LinkList from "./../../components/link-list"
+import Logo from "./../../components/logo"
 
 import styles from "./styles.module.scss"
 
-const Header = () => (
-  <header className={styles.root}>
-    This is the header, and it has <Link to="/">a link that will get you home</Link>.
-    <hr />
-  </header>
-)
+const Header = ({ main_navigation, top_navigation }) => {
+  const [menuIsOpen, setMenu] = useState(false)
+
+  const smallScreenMenuClick = () => {
+    setMenu(!menuIsOpen)
+
+    if (menuIsOpen === false) {
+      document.body.style.overflow = "hidden"
+    } else {
+      document.body.style.removeProperty("overflow")
+    }
+  }
+
+  const classes = classNames(styles.header, {
+    [styles.navigation_is_showing]: menuIsOpen
+  })
+
+  return (
+    <header className={classes}>
+      <div>
+        <div className={styles.logo_container}>
+          <Link className={styles.logo} to="/">
+            <Logo name="ample" />
+          </Link>
+
+          <button className={styles.menu_button} onClick={smallScreenMenuClick}>
+            <span>Menu</span>
+            {!menuIsOpen ? <Icon name="bars" /> : <Icon name="close" />}
+          </button>
+        </div>
 
-Header.propTypes = {}
+        <div className={styles.navigation}>
+          <LinkList className={styles.top_navigation} links={top_navigation} />
+          <LinkList className={styles.main_navigation} links={main_navigation} />
```

Comment:
This is part of the reason I don't love the link list component. There are styles in three places for this particular implementation:

1. Global defaults
2. The link list component
3. This component

If it's almost always going to be customized, it seems like it might actually be extra work to have a component.